### ISSUE TITLE
Document on how end the connection correctly

### ISSUE THREAD

**Issue by xPaw** [kiwiirc/irc-framework#21]
Do you have to send a quit? Do you need to check if connection exists and is connected?

Maybe this should be a proper method.

**Comment by xPaw** [comment]
Doing `irc.quit` seems to crash:

```
events.js:146
      throw err;
      ^

Error: Uncaught, unspecified "error" event. ([object Object])
    at IrcClient.emit (events.js:144:17)
    at irc-framework\src\client.js:136:20
    at next (middleware-handler\index.js:42:18)
    at MiddlewareHandler.handle (middleware-handler\index.js:64:3)
    at IrcCommandHandler.<anonymous> (irc-framework\src\client.js:130:34)
    at emitTwo (events.js:87:13)
    at IrcCommandHandler.emit (events.js:172:7)
    at IrcCommandHandler.emit (irc-framework\src\commands\handler.js:82:23)
    at IrcCommandHandler.<anonymous> (irc-framework\src\commands\handlers\generics.js:134:18)
    at IrcCommandHandler.dispatch (irc-framework\src\commands\handler.js:49:37)
    at IrcCommandHandler._write (irc-framework\src\commands\handler.js:35:10)```

**Comment by prawnsalad** [comment]
The error is because the IRC server actually sends an ERROR to close the connection after a QUIT. If you handle IRC errors (`client.on('error', ...)`) then that error won't be thrown to the process as an unhandled/uncaught error (nodes behaviour). This may change by the framework itself adding an error listener to get around that.

For closing the connection, `client.quit([quit message])` is the correct way to go about it. I've just published version 1.0.5 that will close the socket connection once QUIT has been sent - no longer waiting for the server to close it.

**Comment by xPaw** [comment]
Did you ship 1.0.5 with `console.log` somewhere by mistake?

Also, I handle `socket error` and `irc_error`, what is `error` supposed to report?
